pkg/frontend/asyncoperationresult_get_test.go

```diff
@@ -0,0 +1,272 @@
+							StartTime:                mockOpStartTime,
+							EndTime:                  &mockOpEndTime,
+							Error: &api.CloudErrorBody{
+								Code:    api.CloudErrorCodeInternalServerError,
+								Message: "Some error.",
+							},
+						},
+						OpenShiftCluster: clusterDoc.OpenShiftCluster,
+					}, nil)
+
+				openshiftClusters.EXPECT().Get(gomock.Any(), mockClusterDocKey).
+					Return(clusterDoc, nil)
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse: func() *v20191231preview.OpenShiftCluster {
+				return &v20191231preview.OpenShiftCluster{
+					ID:   "fakeClusterID",
+					Name: "resourceName",
+					Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+				}
+			},
+		},
+		{
+			name: "operation and cluster exist in db - final result is available with no content",
+			mocks: func(openshiftClusters *mock_database.MockOpenShiftClusters, asyncOperations *mock_database.MockAsyncOperations) {
+				asyncOperations.EXPECT().
+					Get(gomock.Any(), mockOpID).
+					Return(&api.AsyncOperationDocument{
+						ID:                  mockOpID,
+						OpenShiftClusterKey: mockClusterDocKey,
+						AsyncOperation: &api.AsyncOperation{
+							ID:                       "fakeOpPath",
+							Name:                     mockOpID,
+							InitialProvisioningState: api.ProvisioningStateCreating,
+							ProvisioningState:        api.ProvisioningStateFailed,
+							StartTime:                mockOpStartTime,
+							EndTime:                  &mockOpEndTime,
+							Error: &api.CloudErrorBody{
+								Code:    api.CloudErrorCodeInternalServerError,
+								Message: "Some error.",
+							},
+						},
+					}, nil)
+
+				openshiftClusters.EXPECT().Get(gomock.Any(), mockClusterDocKey).
+					Return(nil, nil)
+			},
+			wantStatusCode: http.StatusNoContent,
+		},
+		{
+			name: "operation and cluster exist in db - final result is not yet available",
+			mocks: func(openshiftClusters *mock_database.MockOpenShiftClusters, asyncOperations *mock_database.MockAsyncOperations) {
+				asyncOperations.EXPECT().
+					Get(gomock.Any(), mockOpID).
+					Return(&api.AsyncOperationDocument{
+						ID:                  mockOpID,
+						OpenShiftClusterKey: mockClusterDocKey,
+						AsyncOperation: &api.AsyncOperation{
+							ID:                       "fakeOpPath",
+							Name:                     mockOpID,
+							InitialProvisioningState: api.ProvisioningStateUpdating,
+							ProvisioningState:        api.ProvisioningStateFailed,
+							StartTime:                mockOpStartTime,
+							EndTime:                  &mockOpEndTime,
+							Error: &api.CloudErrorBody{
+								Code:    api.CloudErrorCodeInternalServerError,
+								Message: "Some error.",
+							},
+						},
+					}, nil)
+
+				openshiftClusters.EXPECT().Get(gomock.Any(), mockClusterDocKey).
+					Return(&api.OpenShiftClusterDocument{
+						AsyncOperationID: mockOpID,
+					}, nil)
+			},
+			wantStatusCode: http.StatusAccepted,
```

Comment:
Just realised that we need to test a header here.

Reply:
Updated

Reply:
Updated to test header.

Delete already has a test for this. I updated putorpatch in a separate PR (https://github.com/Azure/ARO-RP/pull/78) where I also want to try to turn on the feature flags for tests.